File: gdscript/src/main/java/gdscript/psi/impl/GdComparisonExImpl.java
package gdscript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import com.intellij.psi.util.PsiTreeUtil;
import gdscript.psi.GdComparisonEx;
import gdscript.psi.GdExpr;
import gdscript.psi.GdOperator;
import gdscript.psi.GdVisitor;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class GdComparisonExImpl extends GdExprImpl implements GdComparisonEx {

  public GdComparisonExImpl(@NotNull ASTNode node) {
    super(node);
  }

  @Override
  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitComparisonEx(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

  @Override
  @NotNull
  public List<GdExpr> getExprList() {
    return PsiTreeUtil.getChildrenOfTypeAsList(this, GdExpr.class);
  }

  @Override
  @NotNull
  public GdOperator getOperator() {
    return notNullChild(PsiTreeUtil.getChildOfType(this, GdOperator.class));
  }

}

File: gdscript/src/main/kotlin/gdscript/psi/impl/GdComparisonExImpl.kt
package gdscript.psi.impl

import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import com.intellij.psi.util.PsiTreeUtil
import gdscript.psi.GdComparisonEx
import gdscript.psi.GdExpr
import gdscript.psi.GdOperator
import gdscript.psi.GdVisitor

class GdComparisonExImpl(node: ASTNode) : GdExprImpl(node), GdComparisonEx {
    override fun accept(visitor: GdVisitor) {
        visitor.visitComparisonEx(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }

    override val exprList: List<GdExpr>
        get() = PsiTreeUtil.getChildrenOfTypeAsList(this, GdExpr::class.java)

    override val operator: GdOperator
        get() = notNullChild(PsiTreeUtil.getChildOfType(this, GdOperator::class.java))
}
